build-scripts/compile_profiles.py

```diff
@@ -15,17 +16,30 @@ def __init__(self, * args, ** kwargs):
 
     def resolve(self, all_profiles):
         if self.resolved:
-            return self.raw_selections
+            return
         if self.extends:
+            if self.extends not in all_profiles:
+                msg = (
+                    "Profile {name} extends profile {extended}, but"
+                    "only profiles {known_profiles} are available for resolution."
+                    .format(name=self.id_, extended=self.extends,
+                            profiles=list(all_profiles.keys())))
+                raise RuntimeError(msg)
             extended_profile = all_profiles[self.extends]
-            extended_selects = extended_profile.resolve(all_profiles)
+            extended_profile.resolve(all_profiles)
 
+            extended_selects = extended_profile.raw_selections
             self.raw_selections.update(extended_selects)
-            self.variables.update(extended_profile.variables)
+
+            updated_variables = dict(extended_profile.variables)
+            updated_variables.update(self.variables)
+            self.variables = updated_variables
```

Comment:
I'm thinking right now that we probably shouldn't be messing with the original variables from `Profile`.
This should be `self.raw_variables = updated_variables`.

And how about changing `raw` to `resolved`?

Reply:
When I look at it again, another thought appears - why to bother with `raw_selections` - the `resolve` method could update selections, so they reflect the extended profile, and they take into account unselections and discard them in the process.
In another words, why not to mess with original attributes, when messing means they are just unified with the extended profile and unselections are applied?

Reply:
You have a point, it should be fine to update the original attributes.

Reply:
I have updated the code to use the existing attributes-only approach.

Reply:
Looks good